api/envoy/type/matcher/v3/filter_state.proto

```diff
@@ -25,5 +27,12 @@ message FilterStateMatcher {
 
     // Matches the filter state object as a string value.
     StringMatcher string_match = 2;
+
+    // Matches the filter state object as a ip Instance.
+    RepeatedCidrRange address_match = 3;
   }
 }
+
+message RepeatedCidrRange {
+  repeated xds.core.v3.CidrRange range = 1;
+}
```

Comment:
`ranges`. And I think this could be independent `AddressMatcher` for future usage.